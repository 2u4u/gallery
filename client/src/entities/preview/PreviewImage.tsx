import { PreviewType } from 'shared/types/index.type';

type Props = {
    preview: PreviewType;
}

function PreviewImage({ preview }: Props) {
    return (
        preview.type === 'image' 
            ? (
                <div className='preview__image' style={{backgroundImage: `url(${preview.url})`}} />
            ) 
            : (
                <div className='preview__icon'>
                    <img src={preview.url} alt={preview.name} className='preview__svg'/>
                </div>
            )
    );
}

export default PreviewImage;
