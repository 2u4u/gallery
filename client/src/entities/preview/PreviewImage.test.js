import renderer from 'react-test-renderer';
import PreviewImage from './PreviewImage';

describe('Preview image component ', () => {
    const preview = {
        type: 'image',
        url: 'https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png',
        name: 'test image',
        file: 'Fake file',
    };
    it('matches snapshot', () => {
        const component = renderer.create(
            <PreviewImage preview={preview} />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
