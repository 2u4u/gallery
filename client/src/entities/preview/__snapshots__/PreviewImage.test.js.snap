// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Preview image component  matches snapshot 1`] = `
<div
  className="preview__image"
  style={
    Object {
      "backgroundImage": "url(https://uae-gallery-test-bucket.s3.eu-central-1.amazonaws.com/cat1.png)",
    }
  }
/>
`;
